### DISCUSSION TITLE
Feature request: Interop with `.jsonl` and `.jsonl.gz`

### DISCUSSION THREAD

**Discussion by aaronsteers** [duckdb/duckdb#5155] [General]
Not to suggest this would be a top priority, but would there be any interest in adding support for `jsonl` (and gzipped `jsonl`) to to the list of supported import/export options here?:

<img width="1014" alt="image" src="https://user-images.githubusercontent.com/18150651/199142235-d801ddf4-5dbf-443c-843d-80d8050ec122.png">

While admittedly not the most efficient data storage format, JSONL has some strengths as an interop option, specifically:

- it is supported by most programming languages,
- it is a common choice for logs because it is read and written serially and therefor is easily parallelized,
- it prioritizes write success and source fidelity over validation,
- it supports JSON Schema validation - while at the same time not being bound by it, and
- unlike CSV, it doesn't require any dialect configuration.

**Comment by Mause** [comment]
Hi, we actually already support reading json lines with the `read_ndjson_objects(filename)` function

**Comment by lnkuiper** [comment]
As said, we can read line-delimited json as strings. However, we cannot read and unpack json into columnar data yet. This is something I will implement at some point, but I don't have a timeline yet
